src/doxygen.cpp

```diff
@@ -2597,6 +2597,15 @@ static MemberDef *addVariableToFile(
               "    variable already found: scope=%s\n",qPrint(md->getOuterScope()->name()));
           addMemberDocs(rootNav,md,def,0,FALSE);
           md->setRefItems(root->sli);
+
+          // If md is not external and current is, update declaration location
+          // to current. In case md is external and current isn't, declaration
+          // and definition location is already correct.
+          if(root->explicitExternal && !md->isExternal())
+          {
+            md->setDefFile(root->fileName,root->startLine,root->startColumn);
```

Comment:
This looks wrong to me. If `root->explicitExternal` is true and `md->isExternal()` is false it means root points to a variable _declaration_ not the single _definition_, and `md` probably already points to the definition. As the name suggests the `setDefFile()` should be passed the definition, not a declaration. Doxygen currently extracts documentation found at the declarations to add to the definition, and only falls back to showing the declaration if no definition has been found. So shouldn't the expression have been `!root->explicitExternal && md->isExternal()` instead?

Reply:
With the (newly added) `setDefFile()` I just followed the current naming, where there's `getDefFile()` for getting declaration location and `getDefBodyFile()` / `setDefBody()` for getting / setting definition location. So, um, yes, it might be confusing, but is (I think) consistent with current state. If you have a better idea how to name it, please tell me, I'll happily update.

Besides that, the correctness is verified with the test :) Without the above, the resulting XML file would show

```xml
<location file="080_declaration_definition_order.cpp" line="11" column="1" 
          bodyfile="080_declaration_definition_order.cpp" bodystart="11" bodyend="11"/>
```

instead of (correct)

```xml
<location file="080_declaration_definition_order.cpp" line="32" column="1" 
          bodyfile="080_declaration_definition_order.cpp" bodystart="11" bodyend="11"/>
```